vortex-array/src/array/sparse/canonical.rs

```diff
@@ -29,75 +31,72 @@ impl IntoCanonical for SparseArray {
                     values.validity(),
                     &indices,
                     self.len(),
-                    self.fill_value()
+                    self.fill_value(),
+                    self.dtype().nullability()
                 )
             })
         }
     }
 }
 
 fn canonicalize_sparse_bools(
-    values_array: BoolArray,
+    values: BoolArray,
     indices: &[usize],
     len: usize,
     fill_value: &ScalarValue,
+    nullability: Nullability,
```

Comment:
this arg is redundant, you can get nullability from `values.dtype().nullability()`

Reply:
hm, I didn't check that sparse dtype is values dtype. I thought nullability can be mismatched